api/models/User.js

```diff
@@ -254,114 +254,103 @@ UserSchema.statics.getLeadersByTimeStamp = function(high) {
-  return this.aggregate([
-    { $match: { "portfolio.category": category }},
-    { $unwind: "$portfolio" },
-    { $match: { "portfolio.category": category }},
-    { $sort: { "portfolio.rank": 1 }},
-    { $project: { _id: 1, picture: 1, "portfolio.rank": 1, username: 1 }},
-    { $limit: 10 },
-  ]).exec();
-};
+// Find the top 10 experts by metric for a given category
+// Metrics available are timeStamp and reps given
+// Returns truncated data for efficiency
+UserSchema.statics.getExpertsByMetricForCategory = function(high, category, metric) {
+  var sorter = {};
+  switch(metric) {
+    case 'timeStamp':
+      sorter[metric] = high;
+      break;
+
+    case 'reps':
+      sorter['categories.reps'] = high;
+      break;
+
+    default:
+      sorter['categories.reps'] = high;
+      break;
+  }
 
-// Get top ranked user ids for a given expert category, descending
-// Ranking done by total reps
-// Only returns the username, picture, rank, and reps for the category
-UserSchema.statics.findTopRankedExperts = function(category) {
   return this.aggregate([
     { $match: { "categories.name": category }},
     { $unwind: "$categories" },
     { $match: { "categories.name": category }},
-    { $sort: { "categories.rank": 1 }},
-    { $project: { _id: 1, picture: 1, "categories.rank": 1, "categories.reps": 1, username: 1 }},
+    { $sort: sorter},
+    { $project: { _id: 1, picture: 1, rank: "$categories.rank", reps: "$categories.reps", username: 1, about: 1 }},
     { $limit: 10 },
   ]).exec();
 };
 
-// Get ranked user ids for a given expert category, descending
-// Ranking done by total reps
-// Only returns the IDs of the users in ranked order
-UserSchema.statics.findRankedExperts = function(category) {
-  return this.aggregate([
-    { $match: { "categories.name": category }},
-    { $unwind: "$categories" },
-    { $match: { "categories.name": category }},
-    { $sort: { "categories.reps": -1 }},
-    { $project: { _id: 1 }},
-  ]).exec();
-};
-
-// Get ranked user ids for a given expert category, decreasing
-// Ranking done by total dividends
-UserSchema.statics.findRankedInvestors = function(category) {
+// Find the top 10 investors by metric for a given category
+// Metrics available are timeStamp, average return, total dividends, and rank
+// Currently used for the Investor Dashboard on the Category Page
+// Returns truncated data for efficiency
+UserSchema.statics.getInvestorsByMetricForCategory = function(high, category, metric) {
+  var sorter = {};
+  sorter[metric] = high;
   return this.aggregate([
     { $match: { "portfolio.category": category }},
     { $unwind: "$portfolio" },
     { $unwind: "$portfolio.investments" },
-    { $match: { "portfolio.category": category } },
-    { $group: { _id: "$_id", dividends: { $sum: "$portfolio.investments.dividend" } }},
-    { $sort: { "dividends": -1 }},
-    { $project: { "_id": 1 } },
+    { $match: { "portfolio.category": category }},
+    { $group:
```

Comment:
Grouping requires fields to be inside an object. we put all the old stuff we want to keep inside the _id json block and then unpack it in the $project.